gnd/src/main/java/com/google/android/gnd/model/form/Element.java

```diff
@@ -30,6 +31,21 @@
     FIELD
   }
 
+  public int getIndex() {
+    switch (getType()) {
+      case FIELD:
+        // Fall back to hash code id on missing index; the order won't necessarily be correct
+        // but at least it will be stable.
+        return Optional.ofNullable(getField().getIndex())
+            .orElseGet(() -> getField().getId().hashCode());
+      case UNKNOWN:
+        // Intentional fall-through.
+      default:
+        // Fall back for unknown / bad index.
+        return hashCode();
```

Comment:
What if we make index a required field? If the online form generator always adds an index then there would be no need to take care of adding this attribute manually. Wdyt?

Reply:
Good idea. Let's handle missing `index` in remote closer to the source: eff2520c74fd4db0ac66837d5479965f1b9c40ef.